controllers/application_controller.go

```diff
@@ -244,14 +244,15 @@ func (r *ApplicationReconciler) reconcileHelmReleaseStatus(ctx context.Context,
 		apimeta.RemoveStatusCondition(&application.Status.Conditions, v1.PodReady)
 		return false, nil
 	}
+	helmReadyStatusNotReconciled := true
 	for _, condition := range hr.GetConditions() {
 		apimeta.SetStatusCondition(&application.Status.Conditions, condition)
 		if condition.Type == meta.ReadyCondition && condition.Reason == v2beta1.ReconciliationSucceededReason {
 			apimeta.RemoveStatusCondition(&application.Status.Conditions, v1.PodReady)
-			return false, nil
+			helmReadyStatusNotReconciled = false
```

Comment:
should it do a `break` ?

Reply:
No, all conditions on HR should be on App object + App object's own condition (if applicable). If the HR is not ready, overwhelm will check the pod's status and add it to App object's condition. 

The issue was that if HR was ready, then the early return caused the App object to miss out on "TestSucceeded" / "UpgradeSucceeded" condition.